src/fsharp/NameResolution.fs

```diff
@@ -3886,9 +3892,20 @@ let rec ResolvePartialLongIdentPrim (ncenv: NameResolver) (nenv: NameResolutionE
                |> NameMap.range
                |> List.filter (function Item.ActivePatternCase _v -> true | _ -> false)
 
-       let moduleAndNamespaceItems = 
-           nenv.ModulesAndNamespaces(fullyQualified)
-           |> NameMultiMap.range 
+       let moduleAndNamespaceItems =
+           let moduleOrNamespaceRefs = 
+               nenv.ModulesAndNamespaces fullyQualified
+               |> NameMultiMap.range
+           
+           if isNil moduleOrNamespaceRefs then [] else
+           let ilTyconNames =
+               let hashSet = HashSet()
```

Comment:
This is much better. But, it looks to me there is some duplication. Maybe bring it out into a new function?

e.g.
```fsharp
let getILTyconNamesByTyconRefs (tyconRefs: TyconRef list) =
    let hashSet = HashSet()
    for tycon in tyconRefs do
         if tycon.IsILTycon then 
             hashSet.Add tycon.DisplayName |> ignore
    hashSet

let getILTyconNamesByTycons (tycons: Tycon list) =
    let hashSet = HashSet()
    for tycon in tycons do
         if tycon.IsILTycon then 
             hashSet.Add tycon.DisplayName |> ignore
    hashSet
```